cloudant.js

```diff
@@ -43,17 +43,21 @@ function Cloudant(options, callback) {
     }
   }
 
-  if (theurl.match(/^http:/)) {
+  if (creds.outUrl.match(/^http:/)) {
     options.https = false;
   }
 
   debug('Creating Cloudant client with options: %j', options);
+  // If IAM key found in VCAP, add IAM plugin
+  if (creds && creds.iamApiKey) {
+    options.plugins = { iamauth: { iamApiKey: creds.iamApiKey } };
```

Comment:
Thinking about this, are we overwriting any plugins specified by the user here? I think we should be updating the plugin map if it exists.

Reply:
Yes, it probably would have overwritten existing plugins.  I've fixed this in e9365b1.  I've successfully tested the new logic with the the two client constructors below:
```
var cloudant = Cloudant({ vcapServices: JSON.parse(config)});
```
```
var cloudant = Cloudant({ vcapServices: JSON.parse(config), plugins: [ 'promises', { retry: { retryDelayMultiplier: 4 } } ] });
```